BackendApartmentReservation/BackendApartmentReservation/Apartments/ApartmentRepository.cs

```diff
@@ -11,5 +16,44 @@ public ApartmentRepository(DatabaseContext db)
         {
             _db = db;
         }
+
+        public async Task<DbRoomReservation> CreateRoomReservation(DbEmployee employee, DateTime dateFrom, DateTime dateTo)
+        {
+            var roomReservation = new DbRoomReservation();
+            roomReservation.Employee = employee;
+            roomReservation.DateFrom = dateFrom;
+            roomReservation.DateTo = dateTo;
+
+            var availableRooms = new List<DbApartmentRoom>();
+            foreach (DbApartmentRoom room in _db.ApartmentRooms)
```

Comment:
Thats way too complicated.

```
var rooms = await _db.ApartmentRooms.ToListAsync();
availableRooms = rooms.Where(r => r.DateFrom.Date > dateTo.Date || r.DateTo.Date < dateFrom.Date);
```

Reply:
I think you should use >= and <= here, since there wont be any minutes or hours in the DateTime, only full days